From b6080c3e17e348464b025d34d84f5851ac9f3ba3 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 27 Jan 2017 11:38:45 -0800
Subject: [PATCH 4/4] clang: Do not search clang install dir relative ../lib
 dir

This in OE is installed in native sysroot and bin/../lib
contains native libraries not target libs

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 lib/Driver/ToolChains.cpp | 1 -
 1 file changed, 1 deletion(-)

Index: clang/lib/Driver/ToolChains.cpp
===================================================================
--- clang.orig/lib/Driver/ToolChains.cpp
+++ clang/lib/Driver/ToolChains.cpp
@@ -4221,7 +4221,7 @@ Linux::Linux(const Driver &D, const llvm
   // directory ('Dir' below) or the ResourceDir.
   if (StringRef(D.Dir).startswith(SysRoot)) {
     addPathIfExists(D, D.Dir + "/../lib/" + MultiarchTriple, Paths);
-    addPathIfExists(D, D.Dir + "/../" + OSLibDir, Paths);
+//    addPathIfExists(D, D.Dir + "/../" + OSLibDir, Paths);
   }
 
   addPathIfExists(D, SysRoot + "/lib/" + MultiarchTriple, Paths);
@@ -4265,8 +4265,8 @@ Linux::Linux(const Driver &D, const llvm
   // searched.
   // FIXME: It's not clear whether we should use the driver's installed
   // directory ('Dir' below) or the ResourceDir.
-  if (StringRef(D.Dir).startswith(SysRoot))
-    addPathIfExists(D, D.Dir + "/../lib", Paths);
+  //if (StringRef(D.Dir).startswith(SysRoot))
+  //  addPathIfExists(D, D.Dir + "/../lib", Paths);
 
   addPathIfExists(D, SysRoot + "/lib", Paths);
   addPathIfExists(D, SysRoot + "/usr/lib", Paths);
